### PR TITLE
add `Protocol::TRY`

### PR DESCRIPTION
**Pull request by ModProg**
- [x] added Protocol::TRY and call it in `?`
- [x] add tests
- [x] bikeshed the name

### CODE REVIEW

**Comment by ModProg** [comment]
What would the best place to add a test?

Should we keep `TRY` as the name?

I just used the hash calling `.try()` printed, not sure if that is what you
want.

File: crates/rune-core/src/protocol.rs [hunk 1]
```diff
@@ -414,5 +414,19 @@ impl Protocol {
...
+        ///
+        /// Signature: `fn(self) -> Result`.
+        ///
+        /// Note that it uses the `Result` like [`std::ops::Try`] uses
+        /// [`ControlFlow`](std::ops::ControlFlow) i.e., for `Result::<T, E>`
+        /// it should return `Result<T, Result<(), E>>`
+        pub const TRY: Protocol = Protocol {
+            name: "try",
+            hash: 0xe44ad51ace6f0cd6,
+            repr: None,
```
Comment (udoprog) [inline crates/rune-core/src/protocol.rs:428]: repr: Some("value?"),

**Comment by udoprog** [comment]
> I just used the hash calling .try() printed, not sure if that is what you
want.

Commented elsewhere that they're randomly generated. I'm not exactly sure how you did the above, but if it's at risk of conflicting with something else in the future it's better to avoid!
```sh
cargo run --bin rune -- hash --random
```

**Comment by udoprog** [comment]
> What would the best place to add a test?

Sorry, missed this. In the `rune` crate. I moved the tests in here so that they can make use of `pub(crate)` items and compile faster (linking is exceedingly slow on some platforms).

You can see a bunch of tests in `crates/rune/src/tests`. Just remember to add the module in `tests.rs`.

**Comment by ModProg** [comment]
I'll add a test later, already tested it in my project and it works!

**Comment by udoprog** [comment]
Cheers!

**Pull request merged**